packages/app-mobile/services/commands/stateToWhenClauseContext.ts

```diff
@@ -0,0 +1,16 @@
+// This extends the generic stateToWhenClauseContext (potentially shared by
+// all apps) with additional properties specific to the desktop app. So in
+// general, any desktop component should import this file, and not the lib
+// one.
+
+import libStateToWhenClauseContext, { WhenClauseContextOptions } from '@joplin/lib/services/commands/stateToWhenClauseContext';
+import { AppState } from '../../utils/types';
+
+const stateToWhenClauseContext = (state: AppState, options: WhenClauseContextOptions = null) => {
+	return {
+		...libStateToWhenClauseContext(state, options),
+		keyboardVisible: state.keyboardVisible,
```

Comment:
`keyboardVisible` has been added to the global `whenClauseContext` to allow the "hide keyboard" button to be disabled when the keyboard is already hidden.

Note that the "hide keyboard" button is included to allow minimizing the editor keyboard on iOS.